libgeopmd/src/SysfsIOGroup.cpp

```diff
@@ -52,7 +52,7 @@ namespace geopm
     static std::string read_resource_attribute_fd(int fd)
     {
         char buf[SysfsDriver::M_IO_BUFFER_SIZE] = {0};
-        int read_bytes = read(fd, buf, sizeof buf - 1);
+        int read_bytes = static_cast<int>(read(fd, buf, sizeof buf - 1));
```

Comment:
```suggestion
        auto read_bytes = read(fd, buf, sizeof buf - 1);
```
then you can forgo teh static cast on line 60.  Not really a big deal

Reply:
Read returns a ssize_t type (signed).  sizeof (in C++11) returns a std::size_t type (unsigned).  I'm going to have to cast these both to keep the logic we have.